tests/test_z3_visitor.py

```diff
@@ -0,0 +1,54 @@
+import astroid
+import z3
+
+from python_ta.transforms.z3_visitor import Z3Visitor
+
+
+def test_arithmetic_constraints():
+    z3v = Z3Visitor()
+    code = """
+    def f(x: int, y: int, z: float, a):
+        '''
+        Preconditions:
+            - x ** 2 + y ** 2 == z ** 2
+            - x > 0 and y > 0 and z == 0
+        '''
+        pass
+    """
+    mod = z3v.visitor.visit(astroid.parse(code))
+    function_def = mod.body[0]
+    # Declare variables
+    x = z3.Int("x")
+    y = z3.Int("y")
+    z = z3.Real("z")
+    # Construct expected
+    expected = z3.And([x**2 + y**2 == z**2, z3.And([x > 0, y > 0, z == 0])])
+    actual = z3.And(function_def.z3_constraints)
+    solver = z3.Solver()
+    solver.add(actual == expected)
```

Comment:
Okay in terms of testing methodology, I'm okay with this general approach and using the `solver.check()` at the end. *However* to make the testing a bit more granular and robust against the "contradiction" error we talked about in our meeting, please loop over the `z3_constraints` attribute and check each generated constraint separately. (Rather than the current approach, which `And`s all the constraints together.)